contracts/Remittance.sol

```diff
@@ -65,30 +17,32 @@ contract Remittance is Pausable {
         return keccak256(abi.encodePacked(pass1, pass2));
     }
 
-    // Hash matching function allows password parameters to be passed in either order
-    function doHashesMatch(bytes32 pass1, bytes32 pass2, bytes32 _passHash) private pure returns (bool matches) {
-        return (_passHash == createHash(pass1, pass2) ||
-            _passHash == createHash(pass2, pass1));
+    function doHashesMatch(bytes32 recipientPass, bytes32 exchangerPass, bytes32 _passHash) private pure returns (bool matches) {
+        return (_passHash == createHash(recipientPass, exchangerPass));
     }
 
-    function deposit(address recipient, bytes32 password1, bytes32 password2) 
+    function deposit(address _recipient, address _exchanger, bytes32 hashOfTwoPasswords) 
         public 
         onlyOwner // remove for utility
         payable {
 
         require(msg.value > 0, "Must include value to transfer");
+        require(_recipient != address(0), "Recipient address is missing");
```

Comment:
I probably don't actually need these address checks?

Reply:
It is better to keep the `!= 0` checks because this is what happens when the parameters have been omitted.